api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto

```diff
@@ -181,4 +181,10 @@ message TcpProxy {
   // payloads over a shared HTTP/2 tunnel. If this message is absent, the payload
   // will be proxied upstream as per usual.
   TunnelingConfig tunneling_config = 12;
+
+  // The maximum duration of a connection. The duration is defined as a period since a connection
+  // was established. If not set, there is no max duration. When max_connection_duration is reached
+  // the connection will be closed.
+  // Note: not implemented for upstream connections.
+  google.protobuf.Duration max_connection_duration = 13;
```

Comment:
Changes to v2 isn't allowed anymore. Revert this file.

Reply:
I have reverted v2 changes